### PR TITLE
Issue #242: Add blobstore delete endpoint

### PR DESCRIPTION
**Pull request by hsiehby**
I added the endpoint deleteBlob for deleting blobstores with the given id. Also, the documentation has also been modified to include this endpoint.

### CODE REVIEW

File: gateway/services/upload.go [hunk 1]
```diff
@@ -49,6 +50,12 @@ var UploadRoutes = arbor.RouteCollection{
 		"/upload/blobstore/{id}/",
 		alice.New(middleware.IdentificationMiddleware).ThenFunc(GetBlob).ServeHTTP,
 	},
+	arbor.Route{
+		"DeleteBlob",
+		"DELETE",
+		"/upload/blobstore/{name}/", // ?? Is it correct to use {name}?
```
Comment (benpankow) [inline gateway/services/upload.go]: `name` should work fine, but I would go with `id` since that's what we use elsewhere in this endpoint and what you use in the controller code

File: go.mod [hunk 2]
```diff
@@ -15,3 +15,5 @@ require (
 	gopkg.in/go-playground/validator.v9 v9.25.0
 	gopkg.in/mgo.v2 v2.0.0-20180705113604-9856a29383ce
 )
+
+go 1.13
```
Comment (benpankow) [inline go.mod]: was this change intentional/needed?

Comment (hsiehby) [reply go.mod]: I found that this line is automatically added whenever I run the test.
I have removed it.

File: gateway/services/upload.go (hunk 1 above)
Comment (hsiehby) [reply gateway/services/upload.go]: Okay, I have changed it to id.

**Comment by hsiehby** [comment]
I have removed unnecessary lines and added a test case. Please review my newer commit, thanks!
